### ISSUE TITLE
Closures do modify outer variables

### ISSUE THREAD

**Issue by christophberger** [a8m/go-lang-cheat-sheet#28]
Hi,

I believe the example about Closures is not quite correct. The comment says that the returned outer_var is still 2 because inner() has only modified a copy of it. However, when outer() returns, inner() has not been called yet, and this is why outer_var is 2 at that time. When inner is executed before outer() returns, then outer_var is returned as 101. See this Playground test code: https://play.golang.org/p/e_wktwN6me

Thanks,
Christoph

**Comment by dolmen** [comment]
Here is an example that shows the closure example is incorrect:
```go
package main

import "fmt"

func outer() (int, int) {
    outer_var := 2
    inner := func() int {
        outer_var += 99 // attempt to mutate outer_var from outer scope
        return outer_var // => 101 (but outer_var is a newly redefined
                         //         variable visible only inside inner)
    }
    return inner(), outer_var // => 101, 2 (outer_var is still 2, not mutated by foo!)
}

func main() {
	fmt.Println(outer())
}

```

**Comment by dolmen** [comment]
I confirm that the closure example is completely broken.

**Comment by a8m** [comment]
You're welcome to pr :shipit:

**Comment by a8m** [comment]
Closed via #50
